filter.go

```diff
@@ -372,6 +372,8 @@ func (fs *filterFS) Walk(ctx context.Context, target string, fn gofs.WalkDirFunc
 					return err
 				}
 			}
+
+			dir.calledFn = true
```

Comment:
:eyes: this feels unrelated, were we not setting this before? Was there a case where we were calling the walk target function multiple times? Kinda confused as to why this shows up just here (definitely worth fixing).

Reply:
ya, i moved this line.

Before this pr, the logic went:
- check if the directory is skipped by includePatterns or excludePatterns
- set calledFn to true
- check if the directory is skipped by mapFn
- call fn

now, i only set calledFn to true directly before calling fn